src/main/java/uk/gov/hmcts/cft/idam/testingsupportapi/service/TestingUserService.java

```diff
@@ -13,28 +19,43 @@
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
+import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
+@Slf4j
 public class TestingUserService extends TestingEntityService {
 
-    private final IdamV0Service idamV0Service;
+    private final IdamV2UserManagementApi idamV2UserManagementApi;
 
-    public TestingUserService(IdamV0Service idamV0Service, TestingEntityRepo testingEntityRepo,
+    public TestingUserService(IdamV2UserManagementApi idamV2UserManagementApi, TestingEntityRepo testingEntityRepo,
                               JmsTemplate jmsTemplate) {
         super(testingEntityRepo, jmsTemplate);
-        this.idamV0Service = idamV0Service;
+        this.idamV2UserManagementApi = idamV2UserManagementApi;
     }
 
     /**
      * Create test user.
      *
      * @should create user and testing entity
+     * @should create user and testing entity with roles
+     * @should report if created roles do not match request
      */
     public UserTestingEntity createTestUser(String sessionId, User requestUser, String secretPhrase) {
-        User testUser = idamV0Service.createTestUser(requestUser, secretPhrase);
+        ActivatedUserRequest activatedUserRequest = new ActivatedUserRequest();
+        activatedUserRequest.setPassword(secretPhrase);
+        activatedUserRequest.setUser(requestUser);
+        User testUser = idamV2UserManagementApi.createUser(activatedUserRequest);
```

Comment:
On the Feign end of things it would run through the SpringWebClientErrorDecoder which will throw the appropriate HttpStatusCodeException.

On the response end of things the exception gets caught by CommonExceptionHandler and is turned into an ApiError model.

So a 409 on idamV2UserManagementApi.createUser will result in a 409 as the overall response